common/lib/xmodule/xmodule/tests/test_course_module.py

```diff
@@ -486,19 +487,28 @@ def test_from_json_with_platform_default(self):
             'mock_proctoring_without_rules': {}
         }
     )
-    def test_from_json_with_invalid_provider(self):
+    @ddt.data(True, False)
+    def test_from_json_with_invalid_provider(self, proctored_exams_setting_enabled):
         """
         Test that an invalid provider (i.e. not one configured at the platform level)
         throws a ValueError with the correct error message.
         """
         provider = 'invalid-provider'
         allowed_proctoring_providers = ['mock', 'mock_proctoring_without_rules']
 
-        with pytest.raises(InvalidProctoringProvider) as context_manager:
-            self.proctoring_provider.from_json(provider)
-        expected_error = f'The selected proctoring provider, {provider}, is not a valid provider. ' \
-                         f'Please select from one of {allowed_proctoring_providers}.'
-        assert str(context_manager.value) == expected_error
+        FEATURES_WITH_PROCTORED_EXAMS = settings.FEATURES.copy()
+        FEATURES_WITH_PROCTORED_EXAMS['ENABLE_PROCTORED_EXAMS'] = proctored_exams_setting_enabled
+
+        with override_settings(FEATURES=FEATURES_WITH_PROCTORED_EXAMS):
+            if proctored_exams_setting_enabled:
+                with pytest.raises(InvalidProctoringProvider) as context_manager:
+                    self.proctoring_provider.from_json(provider)
+                expected_error = f'The selected proctoring provider, {provider}, is not a valid provider. ' \
+                    f'Please select from one of {allowed_proctoring_providers}.'
+                assert str(context_manager.value) == expected_error
```

Comment:
error text test seems like overkill since you've already tested that it raises the specific InvalidProctoringProvider error

Reply:
That was there originally. I want to limit my touches